src/services/external/avatar-creator/avatar.creator.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import axios, { AxiosResponse } from 'axios';
+import { LogContext } from '@common/enums';
+
+@Injectable()
+export class AvatarCreatorService {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private readonly logger: LoggerService
+  ) {}
+
+  public generateRandomAvatarURL(firstName: string, lastName?: string): string {
```

Comment:
Not required in this PR, but I've noticed that the generation breaks in case of symbols passed in the firstName/lastName. We should sanitize them before generation.

Reply:
Actually makes a lot of sense to sanitize here.